### ISSUE TITLE
About torch.histc cuda backend

### ISSUE THREAD

**Issue by ZacharyGong** [fangwei123456/pytorch_entropy_loss#1]
Hi,
Thank you for your implementation of calculating the entropy loss using pytorch. It is really useful because (I think) all the DL model for image compression task used some kind of techniques to approximate the entropy loss in the training stage. 
I tried to run this repo on my local enviroment, and I found that the `torch.histc` does not support a cuda tensor as its input. Therefore in the `entropy_loss.py`:
```
p[i] = torch.histc(input[i].view(-1).int(), bins = maxV - minV + 1, min = minV, max = maxV).float() / input[i].numel()
```
it seems not working.

I think it might be changed to:
```
p[i] = torch.histc(input[i].view(-1).int().data.cpu(), bins = maxV - minV + 1, min = minV, max = maxV).float() / input[i].numel()
```
where we might need to put `input[i]` on the "CPU memory".

Also, I guess that you are working on the reproduction of the algorithm mentioned in paper `VARIATIONAL IMAGE COMPRESSION WITH A SCALE HYPERPRIOR`. In the quantification stage, you might need to add a uniform noise as it is mentioned in the paper. 

I'm also working on the reproduction of this paper, I wonder if you could share your progress? Because I faced some issue when implementing the hyperprior module and also my "factorized prior" version cannot match the performance mentioned in the paper.

Looking forward for your reply,

Gong

**Comment by fangwei123456** [comment]
不好意思，现在才注意到这个issue。在较新版本的pytorch（好像1.1之后就可以）中，torch.histc已经支持在cuda上进行运算了。这个计算信息熵损失的仓库已经比较老了，现在我用的是一种新的方法，马上就会在这个库中进行更改。

**Comment by fangwei123456** [comment]
https://github.com/fangwei123456/pytorch_entropy_loss/blob/master/information_entropy.py
